packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -53,6 +53,52 @@ class EinsteinAPI {
         return body
     }
 
+    /**
+     * Given a product source, returns the data that Einstein requires
+     */
+    _destructureProduct(product) {
+        if (product.type && (product.type.master || product.type.variant)) {
+            // handle variants for PDP / viewProduct
+            return {
+                id: product.master.masterId,
+                sku: product.id,
+                altId: '',
+                altIdType: ''
+            }
+        } else if (
+            // handle variants for PLP / viewCategory & viewSearch
+            product.productType &&
+            (product.productType.master || product.productType.variant)
+        ) {
+            return {
+                id: product.productId,
+                sku: product.productId, //TODO: Should this be product.representedProduct.id instead?
+                altId: '',
+                altIdType: ''
+            }
+        } else {
+            // handles non-variants
+            return {
+                id: product.id,
+                sku: '',
+                altId: '',
+                altIdType: ''
+            }
+        }
+    }
+
+    /**
+     * Given a cart item, returns the data that Einstein requires
+     */
+    _destructureItem(item) {
```

Comment:
Continue with naming above, if we were to go with `constructEinsteinProduct`, can we  combine two destructure funcs into one by adding a `type`  arg?

Reply:
I experimented combining this with _constructEinsteinProduct but it felt a little weird since the objects they output are slightly different. Combining the two makes it so that the users of this function will need to know whether they are sending / receiving an item vs a product.